src/agent/src/cdh.rs

```diff
@@ -106,6 +115,73 @@ pub async fn unseal_env(env: &str) -> Result<String> {
     Ok((*env.to_owned()).to_string())
 }
 
+pub async fn unseal_file(path: &str) -> Result<()> {
+    let cdh_client = CDH_CLIENT
+        .get()
+        .expect("Confidential Data Hub not initialized");
+
+    if !Path::new(path).exists() {
+        bail!("sealed secret file {:?} does not exist", path);
+    }
+
+    // Iterate over all entries to handle the sealed secret file.
+    // For example, the directory is as follows:
+    // The secret directory in the guest: /run/kata-containers/shared/containers/21bbf0d932b70263d65d7052ecfd72ee46de03f766650cb378e93852ddb30a54-5063be11b6800f96-sealed-secret-target/:
+    // - ..2024_09_30_02_55_58.2237819815
+    // - ..data -> ..2024_09_30_02_55_58.2237819815
+    // - secret -> ..2024_09_30_02_55_58.2237819815/secret
+    //
+    // The directory "..2024_09_30_02_55_58.2237819815":
+    // - secret
+    for entry in fs::read_dir(path)? {
+        let entry = entry?;
+        let entry_type = entry.file_type()?;
+        if !entry_type.is_symlink() && !entry_type.is_file() {
+            debug!(
+                sl(),
+                "skipping sealed source entry {:?} because its file type is {:?}",
+                entry,
+                entry_type
+            );
+            continue;
+        }
+
+        let target_path = fs::canonicalize(&entry.path())?;
+        info!(sl(), "sealed source entry target path: {:?}", target_path);
```

Comment:
To unpack more on what I want to double check here, in `CCv0` IIRC we unsealed everything to `/run/secrets` , which ensured it was in encrypted memory and I was trying to understand if that logic has changed now. i.e. if I mounted a sealed secret on `/home/root/test` would it always end up unsealed inside `/run`? Thanks

Reply:
> To unpack more on what I want to double check here, in `CCv0` IIRC we unsealed everything to `/run/secrets` , which ensured it was in encrypted memory and I was trying to understand if that logic has changed now. i.e. if I mounted a sealed secret on `/home/root/test` would it always end up unsealed inside `/run`? Thanks

Yes. The sealed secret will always be unsealed in the `/run` directory, specifically in `/run/kata-containers/shared/containers/cid-randomId-test` for your example, and then bind-mounted to the target path `/home/root/test`.

Reply:
> To unpack more on what I want to double check here, in `CCv0` IIRC we unsealed everything to `/run/secrets` , which ensured it was in encrypted memory and I was trying to understand if that logic has changed now. i.e. if I mounted a sealed secret on `/home/root/test` would it always end up unsealed inside `/run`? Thanks

The reason I don’t want to unseal the secret in `/run/secret` is that it might not be cleaned up after the container is deleted.